src/main/java/com/coravy/hudson/plugins/github/GithubLinkAction.java

```diff
@@ -30,4 +35,24 @@ public String getUrlName() {
         return projectProperty.getProjectUrl().baseUrl();
     }
 
+    @SuppressWarnings({"rawtypes"})
+    @Extension
+    public static class Factory extends TransientActionFactory<Job> {
+        @Override
+        public Class<Job> type() {
+            return Job.class;
+        }
+
+        @Override
+        public Collection<? extends Action> createFor(Job j) {
+            GithubProjectProperty prop =
+                    (GithubProjectProperty) ((Job<?, ?>) j).getProperty(GithubProjectProperty.class);
```

Comment:
If you cast to `Job<?,?>` you should get a well-typed return value from `getProperty`. Or you can use `(GithubProjectProperty) j.getProperty(GithubProjectProperty.class)`.